framework/auth/views.py

```diff
@@ -185,6 +191,33 @@ def auth_logout(redirect_url=None):
     return resp
 
 
+def auth_email_logout(token, user):
+    """When a user is adding an email or merging an account, add the email to the user and log them out.
+    """
+    redirect_url = web_url_for('auth_login') + '?existing_user={}'.format(urllib.quote_plus(user.email))
+    try:
+        unconfirmed_email = user.get_unconfirmed_email_for_token(token)
+    except KeyError:
```

Comment:
This seems to catching the wrong error; `get_unconfirmed..` raises either a `InvalidTokenError` or a `ExpiredTokenError`.